src/microbe_stage/OrganelleLayout.cs

```diff
@@ -96,6 +99,20 @@ public bool IsTouchingExistingHex(T organelle)
         return false;
     }
 
+    /// <summary>
+    ///   Returns true if the specified organelle is replacing the last hex of cytoplasm.
+    /// </summary>
+    public bool IsReplacingLast(T organelle)
+    {
+        var replacedOrganelle = GetOrganelleAt(organelle.Position);
+        if (Count == 1 &&
```

Comment:
I have just one more suggestion, this method should begin with:
```
if (Count != 1)
    return;
```
So that  the extra logic doesn't need to run when not needed.

Reply:
&& should be a short-circuit operator in C#, and not run the logic after && if the first part is false.

Reply:
I know. Stylistically I prefer early exit when it's clear that the precondition (there being one organelle) is false.

Reply:
Moved that out in bcd8894